flash/vision/classification/data.py

```diff
@@ -521,3 +558,68 @@ def from_filepaths(
             seed=seed,
             **kwargs
         )
+
+
+class _MatplotlibVisualization(BaseVisualization):
+    """Process and show the image batch and its associated label using matplotlib.
+    """
+    max_cols: int = 4  # maximum number of columns we accept
+
+    @staticmethod
+    def _to_numpy(img: Union[torch.Tensor, Image.Image]) -> np.ndarray:
+        out: np.ndarray
+        if isinstance(img, Image.Image):
+            out = np.array(img)
+        elif isinstance(img, torch.Tensor):
+            out = img.squeeze(0).permute(1, 2, 0).cpu().numpy()
+        else:
+            raise TypeError(f"Unknown image type. Got: {type(img)}.")
+        return out
+
+    def _show_images_and_labels(self, data: List[Any], num_samples: int, title: str):
+        # define the image grid
+        cols: int = min(num_samples, self.max_cols)
+        rows: int = num_samples // cols
+
+        # create figure and set title
+        fig, axs = plt.subplots(rows, cols)
```

Comment:
Raise an exception if matplotlib isn't available here: _MATPLOTLIB_AVAILABLE

Reply:
done, check if that's appropriate